controller/src/beerocks/master/son_management.cpp

```diff
@@ -1803,26 +1803,355 @@ void son_management::handle_bml_message(Socket *sd,
+            LOG(ERROR) << "addClass cACTION_BML_CHANNEL_SCAN_GET_CONTINUOUS_ENABLE_REQUEST failed";
+            break;
+        }
+
+        auto response = message_com::create_vs_message<
+            beerocks_message::cACTION_BML_CHANNEL_SCAN_GET_CONTINUOUS_ENABLE_RESPONSE>(
+            cmdu_tx, beerocks_header->id());
+        if (!response) {
+            LOG(ERROR) << "Failed building message "
+                          "cACTION_BML_CHANNEL_SCAN_GET_CONTINUOUS_ENABLE_RESPONSE !";
+            return;
+        }
+
+        auto radio_mac = request->radio_mac();
+        LOG(DEBUG) << "request radio_mac:" << radio_mac;
+        response->isEnable() = (database.get_channel_scan_is_enabled(radio_mac)) ? 1 : 0;
+
+        //send response to bml
+        message_com::send_cmdu(sd, cmdu_tx);
         break;
     }
     case beerocks_message::ACTION_BML_CHANNEL_SCAN_GET_RESULTS_REQUEST: {
         LOG(TRACE) << "ACTION_BML_CHANNEL_SCAN_GET_RESULTS_REQUEST";
+
+        auto request =
+            beerocks_header
+                ->addClass<beerocks_message::cACTION_BML_CHANNEL_SCAN_GET_RESULTS_REQUEST>();
+        if (!request) {
+            LOG(ERROR) << "addClass cACTION_BML_CHANNEL_SCAN_GET_RESULTS_REQUEST failed";
+            break;
+        }
+
+        // Get information from request
+        auto radio_mac      = request->radio_mac();
+        auto is_single_scan = request->scan_mode() == 1;
+        LOG(DEBUG) << "request radio_mac:" << radio_mac << ", "
+                   << ((is_single_scan) ? "single-scan" : "continuous-scan");
+
+        // Clear flags
+        auto result_status = eChannelScanErrCode::CHANNEL_SCAN_SUCCESS;
+        auto op_error_code = eChannelScanOpErrCode::CHANNEL_SCAN_OP_SUCCESS;
+        uint8_t last       = 0;
+
+        // Get scan statuses
+        auto scan_in_progress = database.get_channel_scan_in_progress(radio_mac, is_single_scan);
+        if (scan_in_progress) {
+            LOG(ERROR) << "Scan is still running!";
+            op_error_code = eChannelScanOpErrCode::CHANNEL_SCAN_OP_SCAN_IN_PROGRESS;
+        }
+
+        auto last_scan_success =
+            database.get_channel_scan_results_status(radio_mac, is_single_scan);
+        if (last_scan_success != eChannelScanErrCode::CHANNEL_SCAN_SUCCESS) {
+            LOG(ERROR) << "Last scan did not finish successfully!";
+            result_status = last_scan_success;
+        }
+
+        // Get results
+        auto scan_results      = database.get_channel_scan_results(radio_mac, is_single_scan);
+        auto scan_results_size = scan_results.size();
+
+        LOG(DEBUG) << "scan_results received for hostap_mac= " << radio_mac << std::endl
+                   << "scan_results_size= " << scan_results_size;
+
+        //Results avaliablity check
+        if (scan_results_size == 0) {
+            LOG(DEBUG) << "no scan results are available";
+            //op_error_code = eChannelScanOpErrCode::CHANNEL_SCAN_OP_SCAN_RESULTS_EMPTY;
```

Comment:
Remove unused code

Reply:
Moved the whole segment to fix an issue with sending CHANNEL_SCAN_RESULTS_EMPTY